### ISSUE TITLE
musicxml.org gives 404 on the dtd

### ISSUE THREAD

**Issue by Jojo-Schmitz** [w3c/musicxml#480]
MuseScore's MusicXML files have this in their header:
`<!DOCTYPE score-partwise PUBLIC "-//Recordare//DTD MusicXML 4.0 Partwise//EN" "http://www.musicxml.org/dtds/partwise.dtd">`
But musicxml.org gives an error 404 on that partwise.dtd (on anything else too).
Initially reported in https://musescore.org/en/node/341860.

**Comment by mscuthbert** [comment]
This might sound ironic, but I believe that the URL in a DOCTYPE definition exists only for uniquely identifying a resource.  It does not have to resolve to an actual address.  That's always been an odd and confusing part of XML in some ways (though in the same boat, Recordare hasn't existed as an independent organization since MakeMusic bought the MusicXML rights and then later released them to W3C) but it's been a part of the specification for some time.

I will try to make it so that 4.1's preferred DOCTYPE resolves to a URL that actually exists, pointing to the .xsd definition in all cases.

And I will try to see if MakeMusic can get musicxml.org back up or if they're willing to transfer the domain name.

**Comment by ahankinson** [comment]
"http://www.musicxml.org/dtds/partwise.dtd" is a URL that points to the DTD, and it _should_ resolve to a remote DTD for validation.

There is no formal MusicXML namespace URI, as [this note](https://www.w3.org/2021/06/musicxml40/listings/musicxml.xsd/) suggests:

"The MusicXML 4.0 DTD has no namespace, so for compatibility the MusicXML 4.0 XSD has no namespace either. Those who need to import the MusicXML XSD into another schema are advised to create a new version that uses "http://www.musicxml.org/xsd/MusicXML" as the namespace."

Also, [this GH issue](https://github.com/w3c/musicxml/issues/266).

**Comment by Jojo-Schmitz** [comment]
"http://www.musicxml.org/xsd/MusicXML" gives an error 404 too...

**Comment by ahankinson** [comment]
Yeah, that one **doesn't** have to resolve -- it's a namespace identifier; it uses assumptions about the uniqueness of domain names to construct a namespace that nobody else can control other than the originating org who owns the domain.

Basically, it looks like a URL but it isn't. It's just a globally unique string.

**Comment by infojunkie** [comment]
The correct way to use `xsltproc`, `xmllint` etc. to resolve DTDs and XSDs is using the `catalog.xml` file that is shipped with the schema here. For example:

`XML_CATALOG_FILES=schema/catalog.xml xsltproc schema/parttime.xsl tests/files/musicxml-tests/xmlFiles/01a-Pitches-Pitches.xml`

`XML_CATALOG_FILES=schema/catalog.xml xmllint --schema schema/musicxml.xsd tests/files/musicxml-tests/xmlFiles/01a-Pitches-P
itches.xml --noout`
